KIPs/kip-2.md

```diff
@@ -0,0 +1,252 @@
+                },
+                "mac": "3355abc00f3fb1ae06821ece9c13359a0da1bd284cf29ffc26a534a40d210f28"
+            }],
+            [],
+            [
+               {
+                   ciphertext: '7c2ad958478c213549fdb9fd7619c6f8c7034618c83e3ab229af6332d9fa53fb',
+                   cipherparams: { iv: '38aa896fc128075425e512f01e4b206c' },
+                   cipher: 'aes-128-ctr',
+                   kdf: 'scrypt',
+                   kdfparams: {
+                       dklen: 32,
+                       salt: 'e7c4605ad8200e0d93cd67f9d82fb9971e1a2763b22362017c2927231c2a733a',
+                       n: 4096,
+                       r: 8,
+                       p: 1,
+                   },
+                   mac: 'e6c3897772916c69f7c778ac2e0e60b786c55f17367c68b086486bd68fea9517',
+               },
+               {
+                   ciphertext: '5a17fe2af445e63ed2cdda6834d030a9391998000941c79318ab49bff092b9e7',
+                   cipherparams: { iv: '38aa896fc128075425e512f01e4b206c' },
+                   cipher: 'aes-128-ctr',
+                   kdf: 'scrypt',
+                   kdfparams: {
+                       dklen: 32,
+                       salt: 'e7c4605ad8200e0d93cd67f9d82fb9971e1a2763b22362017c2927231c2a733a',
+                       n: 4096,
+                       r: 8,
+                       p: 1,
+                   },
+                   mac: '633f91994f33541fbf1c3c3e973e539c12f1dd98f2757f64e3b63de986f367e0',
+               },
+            ]
+        ]
+    }
+    ```
+
+## Rationale
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+The presentation of keys can be an array or an object for a role-based key.
+In this specification, an array is chosen because of the following reasons:
+
+1. Any format has extensibility. If Klaytn introduces a new role, any format can support the new role.
+2. The array format is used when RLP-encoding TxTypeAccountUpdate transactions, so the array format is consistent with this.
+3. Using an object with special strings as keys can have possibilities to use misspelled keys (e.g., "roleTransactin").
+
+## Backwards Compatibility
+<!-- All KIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The KIP must explain how the author proposes to deal with these incompatibilities. KIP submissions without a sufficient backwards compatibility treatise may be rejected outright. The authors should answer the question: "Does this KIP require a hard fork?" -->
+Keystore v3 should also be supported along with the Klaytn keystore format.
```

Comment:
This line means that "Klaytn Keystore Format v4" doesn't have backward compatibility, right?

Reply:
Yes. I will discuss more about that to make it clear.